app/scripts/controllers/authentication/auth-snap-requests.ts

```diff
@@ -0,0 +1,32 @@
+import { SnapId } from '@metamask/snaps-sdk';
+import { HandlerType } from '@metamask/snaps-utils';
+
+const snapId = 'npm:@metamask/message-signing-snap' as SnapId;
+
+export function createSnapPublicKeyRequest() {
+  return {
+    snapId,
+    origin: '',
+    handler: HandlerType.OnRpcRequest,
+    request: {
+      jsonrpc: '2.0',
+      method: 'getPublicKey',
+      params: [],
+      id: 1,
```

Comment:
Can you extract return value as a type and specify it in the function signature ?

Reply:
Yep sure! The SnapController didn't have any nice single internal types. But can recreate or use it.
```
SnapRpcHookArgs & { snapId: SnapId; }
```